crypto3/libs/zk/test/systems/plonk/placeholder/placeholder_dFRI_test_runner.hpp

```diff
@@ -0,0 +1,237 @@
+    using lpc_proof_type = typename lpc_scheme_type::lpc_proof_type;
+    using lpc_placeholder_params_type = nil::crypto3::zk::snark::placeholder_params<circuit_params, lpc_scheme_type>;
+    using policy_type = zk::snark::detail::placeholder_policy<field_type, lpc_placeholder_params_type>;
+    using circuit_type = circuit_description<field_type, placeholder_circuit_params<field_type>>;
+    using fri_type = typename lpc_scheme_type::fri_type;
+    using proof_of_work_type = typename fri_type::grinding_type::output_type;
+    using polynomial_type = typename lpc_scheme_type::polynomial_type;
+    using public_input_type = std::vector<std::vector<typename field_type::value_type>>;
+    using public_preprocessor_type = placeholder_public_preprocessor<field_type, lpc_placeholder_params_type>;
+    using private_preprocessor_type = placeholder_private_preprocessor<field_type, lpc_placeholder_params_type>;
+    using common_data_type = typename public_preprocessor_type::preprocessed_data_type::common_data_type;
+
+    using placeholder_aggregated_proof_type = nil::crypto3::zk::snark::
+                    placeholder_aggregated_proof<field_type, lpc_placeholder_params_type>;
+
+    placeholder_dFRI_test_runner(const circuit_type &circuit1_in, const circuit_type &circuit2_in)
+            : circuit1(circuit1_in), desc1(circuit1_in.table.witnesses().size(),
+                                        circuit1_in.table.public_inputs().size(),
+                                        circuit1_in.table.constants().size(),
+                                        circuit1_in.table.selectors().size(),
+                                        circuit1_in.usable_rows,
+                                        circuit1_in.table_rows),
+              circuit2(circuit2_in), desc2(circuit2_in.table.witnesses().size(),
+                                        circuit2_in.table.public_inputs().size(),
+                                        circuit2_in.table.constants().size(),
+                                        circuit2_in.table.selectors().size(),
+                                        circuit2_in.usable_rows,
+                                        circuit2_in.table_rows),
+              constraint_system1(circuit1_in.gates, circuit1_in.copy_constraints, circuit1_in.lookup_gates, circuit1_in.lookup_tables),
+              constraint_system2(circuit2_in.gates, circuit2_in.copy_constraints, circuit2_in.lookup_gates, circuit2_in.lookup_tables),
+              assignments1(circuit1_in.table),
+              assignments2(circuit2_in.table),
+              table_rows_log(std::log2(circuit1_in.table_rows)),
+              fri_params(1, table_rows_log, placeholder_test_params::lambda, 4) {
+    }
+
+    // The code in this function generates an aggregated dFRI proof using the provided 2 circuits.
+    // It is very similar to the code in the proof-producer. Should something change in the algorithm,
```

Comment:
The code of this function should match that in the proof-producer. I believe it matches :)